Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE41)' during 'Importation'

Failing in native AOT outerloops.

```
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Generating native code
  ILC: /__w/1/s/src/coreclr/jit/gentree.cpp:26408
  ILC: Assertion failed 'compIsaSupportedDebugOnly(InstructionSet_SSE41)' in 'System.Numerics.Vector`1[double]:System.Runtime.Intrinsics.ISimdVector<System.Numerics.Vector<T>,T>.IsInteger(System.Numerics.Vector`1[double]):System.Numerics.Vector`1[double]' during 'Importation' (IL size 7; hash 0x24264ea0; FullOpts)
  
/__w/1/s/artifacts/bin/coreclr/linux.x64.Checked/build/Microsoft.NETCore.Native.targets(316,5): error MSB3073: The command ""/__w/1/s/artifacts/bin/coreclr/linux.x64.Checked/x64/ilc/ilc" @"/__w/1/s/artifacts/obj/System.Numerics.Vectors.Tests/Release/net10.0/native/System.Numerics.Vectors.Tests.ilc.rsp"" exited with code 134. [/__w/1/s/src/libraries/System.Numerics.Vectors/tests/System.Numerics.Vectors.Tests.csproj]
##[error]artifacts/bin/coreclr/linux.x64.Checked/build/Microsoft.NETCore.Native.targets(316,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command ""/__w/1/s/artifacts/bin/coreclr/linux.x64.Checked/x64/ilc/ilc" @"/__w/1/s/artifacts/obj/System.Numerics.Vectors.Tests/Release/net10.0/native/System.Numerics.Vectors.Tests.ilc.rsp"" exited with code 134.
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=922740&view=logs&jobId=7e7cde7f-4669-5f41-976b-370a74291e60&j=7e7cde7f-4669-5f41-976b-370a74291e60&t=9ee39664-b162-579c-6968-bf80271a4c1d

Cc @tannergooding

FYI I hit this issue trying to repro, will take a look at it after I work on the above.
```
/home/andya/repos/runtime0/artifacts/bin/coreclr/linux.x64.Checked/build/Microsoft.NETCore.Native.targets(316,5): error MSB3073: The command ""/home/andya/repos/runtime0/artifacts/bin/coreclr/linux.x64.Checked/x64/ilc/ilc" @"/home/andya/repos/runtime0/artifacts/obj/System.Drawing.Primitives.Tests/Release/net10.0/native/System.Drawing.Primitives.Tests.ilc.rsp"" exited with code 134. [/home/andya/repos/runtime0/src/libraries/System.Drawing.Primitives/tests/System.Drawing.Primitives.Tests.csproj]
```

--- a/src/coreclr/jit/hwintrinsicxarch.cpp
+++ b/src/coreclr/jit/hwintrinsicxarch.cpp
@@ -2925,11 +2925,17 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 1);
 
-            if ((simdSize != 32) || compOpportunisticallyDependsOn(InstructionSet_AVX2))
+            if ((simdSize == 16) && !compOpportunisticallyDependsOn(InstructionSet_SSE41))
             {
-                op1     = impSIMDPopStack();
-                retNode = gtNewSimdIsIntegerNode(retType, op1, simdBaseJitType, simdSize);
+                break;
             }
+            if ((simdSize == 32) && !compOpportunisticallyDependsOn(InstructionSet_AVX2))
+            {
+                break;
+            }
+
+            op1     = impSIMDPopStack();
+            retNode = gtNewSimdIsIntegerNode(retType, op1, simdBaseJitType, simdSize);
             break;
         }
 

